### PR TITLE
Take timezone offset into account when parsing DateTime

### PR DESCRIPTION
**Pull request by TBonnin**
Previous version of DateTime deserialiser was not taking into account timezone offset

### CODE REVIEW

File: facia-json/src/main/scala/com/gu/facia/client/json/JodaFormat.scala [hunk 1]
```diff
@@ -3,7 +3,8 @@ package com.gu.facia.client.json
 import org.joda.time.DateTime
 import org.joda.time.format.ISODateTimeFormat
 import play.api.libs.json._
-import scala.util.control.Exception.nonFatalCatch
+
+import scala.util.control.Exception.allCatch
```
Comment (TBonnin) [inline facia-json/src/main/scala/com/gu/facia/client/json/JodaFormat.scala]: not related to the PR but a improvement nonetheless

File: facia-json/src/main/scala/com/gu/facia/client/json/JodaFormat.scala [hunk 2]
```diff
@@ -20,14 +21,15 @@ object JodaReads {
...
+    val dateFormat = ISODateTimeFormat.dateTimeParser().withOffsetParsed()
+
+    def parse(s: String) = allCatch[DateTime] opt (DateTime.parse(s, dateFormat))
 
     def reads(json: JsValue): JsResult[DateTime] = json match {
       case JsNumber(d) => JsSuccess(new DateTime(d.toLong))
       case JsString(s) => parse(s) match {
         case Some(d) => JsSuccess(d)
-        case _ => JsError(JsPath(), s"error.expected.jodadate.format ${dateFormat.toString}")
+        case _ => JsError(JsPath(), s"error.unexpected.date.format. Date: '$s'")
```
Comment (TBonnin) [inline facia-json/src/main/scala/com/gu/facia/client/json/JodaFormat.scala]: not related neither but a better error message that show the string that cannot be parsed

File: facia-json/src/main/scala/com/gu/facia/client/json/JodaFormat.scala [hunk 3]
```diff
@@ -20,14 +21,15 @@ object JodaReads {
 
   implicit val JodaDateTimeReads : Reads[DateTime] = new Reads[DateTime] {
 
-    val dateFormat = ISODateTimeFormat.dateOptionalTimeParser
-    def parse(s: String) = nonFatalCatch[DateTime] opt (DateTime.parse(s, dateFormat))
+    val dateFormat = ISODateTimeFormat.dateTimeParser().withOffsetParsed()
```
Comment (TBonnin) [inline facia-json/src/main/scala/com/gu/facia/client/json/JodaFormat.scala]: This is the fix `.withOffsetParsed()`

**Comment by janua** [comment]
Thanks for the explanation :+1:

**Pull request merged**